src/main/java/edu/ucla/library/iiif/auth/handlers/AbstractAccessTokenHandler.java

```diff
@@ -95,6 +95,7 @@ public AccessCookieService getAccessCookieService() {
     }
 
     @Override
+    @SuppressWarnings("PMD.CognitiveComplexity")
```

Comment:
I happen to agree with PMD that this version of the conditional is (at least marginally) more cognitively complex. I wonder if it's due to the `!=` and having to perform an extra "negation" step. I can live with it as is, but I would argue that we should consider sometimes rejecting this sort of code rewriting by the IDE.

Reply:
I'm not sure we can say this alone is what made it trip the cognitively complex check(?) That whole method seems like it could be simplified (with perhaps the negation step being the straw that broke PMD's back). I didn't intend for the additions of that `SuppressWarnings` annotation to mean that we shouldn't return to these places, later, for simplification. I look at them as flags for areas we do need to return to in future refactorings.

I'm not sure how I feel about this particular Eclipse "simplification" -- I can see both sides of the argument on it. I'm fine with us looking at the Eclipse simplifications, though, and making our own decisions about whether we want to use them or not (it's turned on by default with the newer Eclipse, but we can turn it off again too).

Reply:
Re: "consider _sometimes_ rejecting" ... I think we should make the decision whether we want that type of simplification or not (and turn it off if we don't). I'd prefer consistency over making an on-the-fly decision on a case-by-case basis. Doing it on a case-by-case basis seems too randomly subjective to me.

Reply:
The reason I didn't move the `onFailure` callback to its own private method is because it references values defined outside like `isBrowserClient`, so it seemed easier to lump together. But I could see it being split out as an improvement.

Ok, that's fair -- it is getting into subjective territory.

Reply:
I've created a ticket for us to look at these types of automatic cleanups that Eclipse can perform and make a collective subjective decision on whether we want them or not. :-)